test/jasmine/tests/hover_spikeline_test.js

```diff
@@ -404,6 +404,29 @@ describe('spikeline hover', function() {
         .then(done);
     });
 
+    it('correctly select the closest bar even when setting spikedistance to -1', function(done) {
+        var mock = require('@mocks/bar_stack-with-gaps');
+        var mockCopy = Lib.extendDeep({}, mock);
+        mockCopy.layout.xaxis.showspikes = true;
+        mockCopy.layout.yaxis.showspikes = true;
+        mockCopy.layout.spikedistance = -1;
+
+        Plotly.newPlot(gd, mockCopy)
```

Comment:
Non-blocking:
We could us `newPlot` > `plot` here.

Reply:
Is that desirable? `plot` is not part of the API we want anyone using, so I'd kind of rather not use it in tests either, even if it's a little faster.

Reply:
Thanks for the note. Resolving...
And please feel free to open an issue if you prefer to chage similar cases in all the tests.

Reply:
I wouldn't bother converting the rest unless there are problems. Sometimes there's a good reason to go for the faster one, in particular I seem to recall a lot of overhead to `newPlot` in WebGL tests? But that overhead comes with more assurance that there's no cross-test interaction, which has lead to hard-to-debug errors in the past. @etpinard any words of wisdom to add here?